Created a resource folder and created a python script to create one json

matiasbavera opened the pull request:
### What's new
* Created a resource folder where the icons of robots, dispensers, etc. will be stored. It also is stored configurations of available places by fleets and dispensers in each place.
* Created a script to generate one JSON with all of the configurations. 

### Discussion
* Is the location of the folder ok? Or is it better to move it to assets? 
* Is this structure ok?
* I created the bundler script in python to keep the as a fewer programming language as possible in this project
![image](https://user-images.githubusercontent.com/11761240/89233142-02be9f80-d5b7-11ea-8ad3-8015c7ec100b.png)
* I put some examples (maybe they are not valid but they can be used to discuss the structure) 
* Since we have one model of robot per fleet. I think there is no need to create folders in this format: `robots/fleetname/modelname/icon.png`. But if in the future we can have more than one model per fleet, makes total sense.

### Context
We need to set icons provided by different vendors in the Romi-dashboard project. For this, we will need to store the icons and paths somewhere.
These icons cannot be directly in the Romi-dashboard project because the project is going to have several icons/images that should not be inside the Romi-dashboard project. Also, Romi-dashboard should be used in various places with different robots (and vendors) so we shouldn't have a fixed path for your icons because it can vary.
So the idea is to have these icons in a central place, which would be this project, and provide Romi-dashboard with a file with the path to the images.

Romi-dashboard needs to get the icons and have a config file (JSON) with the image paths. Romi-dashboard will copy the icons that it needs in a pre-build process and the same project will serve the icons with the configuration file.


CC: @Yadunund

Yadunund commented on resources/robots/magni/magni.json:
```diff
@@ -0,0 +1,19 @@
+        "icons":{
+            "magni": "/robots/magni/magni.png"
+        }, 
+        "places":[ "supplies",
```
Correct me if I'm remembering wrong, but did we agree that the items in `places` will be a tuple of the place's name along with the dispenser name if any?

matiasbavera replied:
Thanks for the feedback!
Yes, but I just wanted to show you this to see if there was any possibility of separating concepts. But it actually seems to be pretty correlated concepts. I feel like I'm missing something so let me check this with you if this affirmation is correct:
 just certain types (fleets) of robots can use the dispensers and that's why we should add the places and the dispensers inside the fleet and no as a general place-dispenser file.

Following the previous logic and as we discussed before, I'll follow this structure 
```
[{
place:NameOfPlace, 
dispensers:[dispenser1, dispenser2] 
},{}]
```

Yadunund replied:
Agreed.

Yadunund commented on resources/robots/magni/magni.json:
```diff
@@ -0,0 +1,19 @@
+{
+    "magni":{
```
Since we have multiple demo worlds in this repository with a different set of places, I'm wondering if we should add a `building_name` field that is the parent of "magni". So we will have two such fields, 1) `office` 2)airport_terminal

matiasbavera replied:
You're right! I'll do that, Romi-dashboard will have a then a configuration file to know where to point at.

matiasbavera replied:
Hi @Yadunund , I need to ask a question here, should an implementation need to support multiple worlds? For example, can the airport_terminal world contain an office world with different maps and different fleets and robots?

In case not:
I had been thinking and I realize that in the end game we will be referencing just one project (or world) in this case `office` or `airport_terminal`. I think these two worlds' resources should be on different folders and it should not have one file unifying its configurations. Each one should have its own main JSON configuration file and RomiDashboard should clone the world folder that we want to show.

Let me know what you think about it

Yadunund replied:
You're right, separating the main json files for different worlds is a better way to manage the files.

Yadunund commented on resources/dispensers/coke/dispensers.json:
```diff
@@ -0,0 +1,11 @@
+{
```
I think we can eliminate this file. See comment on dispenser and places below.

matiasbavera replied:
Sure thing!

Yadunund replied:
The robot icons could be referenced in the `magni.json` file? Unless you're referring to icons of dispensers?

matiasbavera replied:
Hi @Yadunund! Yes, sorry for being unclear, I'm referring to dispensers' icons. 
My idea is to have something like this

`dispernsers.json`
```
{
    "coke":{
        "icons":{
            "coke": "/robots/magni/magni.png"
        }, 
    }
}
```
A file just with the icons. Because, as you said, the places and its dispenser are already on each <fleet>.json.

I'll remove this file for now because we don't have any icons for the dispensers. I'm writing this here for just a reminder of how the structure should be when we want to create it again.

Yadunund replied:
Ah okay a file for the dispenser icon makes sense. Thanks for letting me know.

Yadunund commented on resources/robots/fleetA/fleetA.json:
```diff
@@ -0,0 +1,8 @@
+{
```
I know this is just an example file but we should eventually re-use this for the `mir100` fleet instead of `fleetA`

matiasbavera replied:
Yeah, I'll change the name to mir100

Yadunund commented:
Could we also rename the `resources` folder to `rmf_dashboard_resources`?

matiasbavera commented:
> Could we also rename the `resources` folder to `rmf_dashboard_resources`?

Sure thing

matiasbavera commented:
Related to https://github.com/osrf/romi-dashboard/pull/66

Yadunund reviewed (changes requested):
Thanks for addressing all the issues so far @matiasbavera . One last thing to take care of is renaming all instances of `MiR100` to `DeliveryRobot` and `Magni` to `TinyRobot`. This is in line with out internal push to avoid using proprietary names. PR #115 is taking care of the renaming in other places in this repo.

matiasbavera commented:
> Thanks for addressing all the issues so far @matiasbavera . One last thing to take care of is renaming all instances of `MiR100` to `DeliveryRobot` and `Magni` to `TinyRobot`. This is in line with out internal push to avoid using proprietary names. PR #115 is taking care of the renaming in other places in this repo.

Oh! Oh right! makes sense, I'll do that on RomiDashboard too.

matiasbavera commented:
Done!
